spec.html

```diff
@@ -426,12 +427,11 @@ <h1>IsValidIntegerIndex ( _O_, _index_ )</h1>
     <p>The abstract operation IsValidIntegerIndex takes arguments _O_, _getBufferByteLength_, and _index_. It performs the following steps when called:</p>
     <emu-alg>
       1. Assert: _O_ is an Integer-Indexed exotic object.
-      1. Assert: Type(_index_) is Number.
-      1. If ! IsInteger(_index_) is *false*, return *false*.
+      1. <del>If IsDetachedBuffer(_O_.[[ViewedArrayBuffer]]) is *true*, return *false*.</del>
+      1. If ! IsIntegralNumber(_index_) is *false*, return *false*.
       1. If _index_ is *-0*<sub>𝔽</sub>, return *false*.
       1. <ins>Let _getBufferByteLength_ be ! MakeIdempotentArrayBufferByteLengthGetter(~Unordered~).</ins>
       1. <ins>NOTE: Bounds checking is not a synchronizing operation when _O_'s backing buffer is a growable SharedArrayBuffer.</ins>
-      1. <ins>If IsIntegerIndexedObjectOutOfBounds(_O_, _getBufferByteLength_) is *true*, return *false*.</ins>
       1. If ℝ(_index_) &lt; 0 or ℝ(_index_) &ge; <del>_O_.[[ArrayLength]]</del><ins>IntegerIndexedObjectLength(_O_, _getBufferByteLength_)</ins>, return *false*.
```

Comment:
Please add an assert and a note:

`1. Assert: IsDetachedBuffer(_O_.[[ViewedArrayBuffer]]) is *false*.`
`1. NOTE: When the buffer is detached, IntegerIndexedObjectLength returns 0.`